fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneLockFailureTest.java

```diff
@@ -0,0 +1,261 @@
+
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.TestRecordsTextProto;
+import com.apple.foundationdb.record.lucene.directory.FDBDirectory;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.IndexOptions;
+import com.apple.foundationdb.record.provider.common.text.AllSuffixesTextTokenizer;
+import com.apple.foundationdb.record.provider.foundationdb.FDBExceptions;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTestBase;
+import com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils;
+import com.apple.foundationdb.record.query.expressions.Query;
+import com.apple.foundationdb.record.query.plan.QueryPlanner;
+import com.apple.foundationdb.record.util.pair.Pair;
+import com.apple.foundationdb.subspace.Subspace;
+import com.apple.foundationdb.tuple.Tuple;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.store.LockObtainFailedException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+
+import static com.apple.foundationdb.record.lucene.LuceneIndexOptions.INDEX_PARTITION_BY_FIELD_NAME;
+import static com.apple.foundationdb.record.lucene.LuceneIndexOptions.INDEX_PARTITION_HIGH_WATERMARK;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTest.ENGINEER_JOKE;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTest.complexPartitionedIndex;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.SIMPLE_TEXT_SUFFIXES;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createComplexDocument;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createSimpleDocument;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.openRecordStore;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.COMPLEX_DOC;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.SIMPLE_DOC;
+
+class LuceneLockFailureTest extends FDBRecordStoreTestBase {
+
+    private static final Index SIMPLE_INDEX = SIMPLE_TEXT_SUFFIXES;
+
+    protected static final Index COMPLEX_PARTITIONED = complexPartitionedIndex(Map.of(
+            IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME,
+            INDEX_PARTITION_BY_FIELD_NAME, "timestamp",
+            INDEX_PARTITION_HIGH_WATERMARK, "10"));
+
+    @Test
+    void testAddDocument() throws IOException {
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, SIMPLE_INDEX);
+            grabLockExternally(SIMPLE_INDEX, context);
```

Comment:
I feel like we should grab the lock in a separate transaction. I can't think of any situation where that should matter, but it much more accurately represents real use cases.

Reply:
Done